app/src/components/main/modal/modalLayout/modalLayout.jsx

```diff
@@ -84,13 +84,23 @@ export class ModalLayout extends Component {
     }
   };
   onClickModal = (e) => {
-    !this.modal.contains(e.target) && !this.props.stopOutsideClose && this.closeModal();
+    const { closeConfirmation } = this.props;
+
+    if (!this.modal.contains(e.target)) {
+      if (!closeConfirmation) {
+        this.closeModal();
+      } else {
+        closeConfirmation.withCheckbox
+          ? this.setState({ showConfirmation: true })
+          : this.setState({ showConfirmation: true, closeConfirmed: true });
+      }
+    }
   };
   onClickOk = () => {
-    const { okButton, multiActionOkButton } = this.props;
+    const { okButton, customButton } = this.props;
 
     okButton && okButton.onClick(this.closeModal);
-    multiActionOkButton && multiActionOkButton.onClick(this.closeModal);
+    customButton && customButton.onClick(this.closeModal);
```

Comment:
If we have a custom button and an ok button, one button will trigger both handlers. That's not what we want. Let's just pass `closeModal` to a custom button (see comment above)

Reply:
Done